### ISSUE TITLE
[Meta] A11y keyboard control + focus state issues

### ISSUE THREAD

**Issue by zackkrida** [creativecommons/cccatalog-frontend#1106]
## Problem

There are a number of issues surrounding modals and some other clickable components, some don't have an on-focus indicator, some don't have trap focus and some of the clickable components can't be pressed using the return(enter) key.

## Description
1. Some buttons don't have an on-focus indicator:

![CC Search - Brave 05-08-2020 18_44_31](https://user-images.githubusercontent.com/45792768/89417852-e48aa800-d74c-11ea-97a2-3776437ae8a4.png)
![CC Search - Brave 05-08-2020 18_45_31](https://user-images.githubusercontent.com/45792768/89417970-084dee00-d74d-11ea-88c2-bbe49cb6ce0d.png)

<strike> 2. Modals don't have trap-focus, i.e if you press tab on reaching the end of the modal the focus jumps out of the modal. Also the modal should close if an area outside the modal is clicked.
</strike>

<strike> 3. Some buttons are clickable but cannot be pressed using the return(enter) key:
</strike>

There is also an alignment issue with the focus box in the filter tabs.
The relevant issues in vocabulary are:
1. https://github.com/creativecommons/vocabulary/issues/562
<strike> 2. https://github.com/creativecommons/vocabulary/issues/563 </strike>

## Implementation

- [ ] I would be interested in implementing this feature.
